src/react/projects/spark-core-react/src/SprkStack/SprkStack.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+const SprkStack = (props) => {
+  const {
+    children,
+    itemSpacing,
+    splitAt,
+    additionalClasses,
+    idString,
+    ...other
+  } = props;
+
+  const classNames = classnames('sprk-o-Stack', additionalClasses, {
+    'sprk-o-Stack--tiny': itemSpacing === 'tiny',
+    'sprk-o-Stack--small': itemSpacing === 'small',
+    'sprk-o-Stack--medium': itemSpacing === 'medium',
+    'sprk-o-Stack--large': itemSpacing === 'large',
+    'sprk-o-Stack--huge': itemSpacing === 'huge',
+    'sprk-o-Stack--misc-a': itemSpacing === 'misc-a',
+    'sprk-o-Stack--misc-b': itemSpacing === 'misc-b',
+    'sprk-o-Stack--misc-c': itemSpacing === 'misc-c',
+    'sprk-o-Stack--misc-d': itemSpacing === 'misc-d',
+    'sprk-o-Stack--split@xxs': splitAt === 'extraTiny',
+    'sprk-o-Stack--split@xs': splitAt === 'tiny',
+    'sprk-o-Stack--split@s': splitAt === 'small',
+    'sprk-o-Stack--split@m': splitAt === 'medium',
+    'sprk-o-Stack--split@l': splitAt === 'large',
+    'sprk-o-Stack--split@xl': splitAt === 'huge',
+  });
+
+  return (
+    <div className={classNames} data-id={idString} {...other}>
+      {children}
+    </div>
+  );
+};
+
+SprkStack.defaultProps = {
+  children: '',
+  splitAt: '',
+  itemSpacing: '',
+  idString: '',
```

Comment:
with a default value of `''`, the element will always render `data-id`. If you set the default value to `null`, the `data-id` attribute won't be added to the element unless it has a value.

Reply:
We are going to chat about defaultProps in our next programming hour.